assets/javascripts/discourse/initializers/chat-notification-sounds.js

```diff
@@ -13,13 +19,33 @@ export default {
       return;
     }
 
+    let canPlay = true;
+
+    function playAudio(user) {
+      const audio = new Audio(CHAT_SOUNDS[user.chat_sound]);
+      audio.play().catch(() => {
+        // eslint-disable-next-line no-console
+        console.info(
+          "User needs to interact with DOM before we can play notification sounds"
+        );
+      });
+    }
+
+    function playAudioWithDebounce(user) {
+      if (canPlay) {
+        canPlay = false;
+
+        setTimeout(() => {
```

Comment:
Can you use [debounce](https://api.emberjs.com/ember/release/functions/@ember%2Frunloop/debounce) for this?

Reply:
Should be able to, this is just what Mattermost is doing in their source, I linked that code in our internal chat :)

Reply:
https://github.com/discourse/discourse-chat/pull/470